templates/shared/search/button.tmpl

```diff
@@ -0,0 +1,3 @@
+<!-- Disable (optional) - if search button has to be disabled -->
+<!-- Tooltip (optional) - a tooltip to be displayed on hover -->
+<button class="ui small primary icon button" aria-label="{{ctx.Locale.Tr "search.search"}}" {{with .Tooltip}}data-tooltip-content="{{.}}"{{end}}{{if .Disabled}} disabled{{end}}>{{svg "octicon-search"}}</button>
```

Comment:
What does `with` do and shouldn't that be an `if`?

Reply:
If `.Tooltip` is set, the following code block will be executed and `.` becomes the value of `.Tooltip`. It's basically just a cleaner notation compared to the if-one and I use it pretty often. If you don't like it in Gitea I can also change it to if.

Reply:
I guess `if` would be more common in gitea code. Changing the value of `.` is also often a problem as seen in loops, but I guess in this simple case it's ok.

Reply:
I would like to keep `with` if it's not a blocker for you, as this one-line templates should stay as minimalistic as possible and this notation is simply shorter and easier to understand from my pov

Reply:
I forbid `with` in my JS, but I guess for go template it might be borderline ok 😆

Reply:
I also had to look twice but i find it more elegant.

And if with is used more often people will notice imidiatly :)